prepare_inventory.py

```diff
@@ -0,0 +1,47 @@
+#!/bin/env python3
+
+# This is simple script which converts EDPM inventory from the OpenShift's secret
+# to the very simple inventory file which can be used by the RHOSO break/fix training
+# scripts while preparing training scenarios.
+# Usage of the script:
+# python3 prepare_inventory.py <path_to_the_edpm_inventory.yaml> <path_to_the_output_file.yaml>
+#
+# Edpm inventory file can be get from the OpenShift with command like:
+#
+# oc get secret <secret_name> -o json | jq '.data | map_values(@base64d)' | jq -r '.inventory'
+
+
+import yaml
+import sys
+
+
+if __name__ == '__main__':
+    src_file = sys.argv[1]
+    dst_file = sys.argv[2]
+    if not src_file:
+        print("Yaml file with the RHOSO edpm inventory is required "
+              "to be passed as first argument.")
+        sys.exit(1)
+    if not dst_file:
+        print("Output file is required to be passed as second argument.")
+        sys.exit(1)
+
+    with open(src_file) as src:
+        edpm_inventory = yaml.safe_load(src)
+
+    dest_inventory = {}
+    for group_name, group in edpm_inventory.items():
+        if group_name not in dest_inventory:
+            dest_inventory[group_name] = {'hosts': {}}
+        for hostname, host_options in group.get('hosts').items():
+            canonical_hostname = host_options.get('canonical_hostname')
+            ext_ip = host_options.get('external_ip')
+            ansible_host = canonical_hostname or ext_ip or host_options.get('ansible_host')
+            dest_inventory[group_name]['hosts'][hostname] = {
+                'ansible_user': host_options.get('ansible_user'),
+                'ansible_host': canonical_hostname or ext_ip or
+                                host_options.get('ansible_host')
```

Comment:
you probably wanted to use the `ansible_host` variable here? :)

Reply:
actually no, if there is canonical hostname, I noticed in the lab that it is properly resolved on the student workstation machine in the lab so I wanted to use that one as a first option. This script is very oriented to the trainings lab so that's why I did it that way.

Reply:
but it is used as a first option on L45. It seems this is exactly a copy of what is stored to the ansible_host variable on L45

Reply:
ahh, right. I forgot to update it here :)